plannerbenchmark/postProcessing/metrics.py

```diff
@@ -1,37 +1,59 @@
 import numpy as np
-import abc
+from abc import ABC, abstractmethod
+from typing import Callable
 
 
-def computeDistances(positions, point):
-    return np.linalg.norm(np.add(positions, -point), axis=1)
+def computeDistances(points_a: np.ndarray, points_b: np.ndarray):
+    """Compute the distance between two array of points."""
+    return np.linalg.norm(np.add(points_a, -points_b), axis=1)
 
 
-def computeFks(positions, fks_fun):
-    all_fks = []
-    for pos in positions.T:
-        all_fks.append(np.array(fks_fun(pos)))
-    return np.array(all_fks)
+class Metric(ABC):
+    """Abstract metric to assess the performance of the motion planner.
 
+    Attributes
+    ----------
 
-class Metric(object):
-    def __init__(self, name, measNames, params):
-        self._name = name
-        self._params = params
-        self._measNames = measNames
+    _name: str
+        Name of the metric.
+    _params: dict
+        Additional information and data is passed through the dictonary.
+    _measNames: list
+        List of keys that are needed from the results.
 
-    @abc.abstractmethod
-    def computeMetric(self, data):
+    """
+
+    def __init__(self, name: str, measNames: list, params: dict):
+        self._name: str = name
+        self._params: dict = params
+        self._measNames: list = measNames
+
+    @abstractmethod
+    def computeMetric(self, data: dict):
         return
 
+    def name(self):
```

Comment:
This method appears to be not used.
General remark: Using a leading underscore in a variable's name is only a convention used to indicate that the variable is private, but in practice the variable can be overwritten. Consider using `property` to manage the private read-only attributes as a safer option.

Reply:
Created separate issue for that, #18